File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/WhiteSameNameRefDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.optional.OptionalEntity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of white_same_name_ref. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class WhiteSameNameRefDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final WhiteSameNameRefDbm _instance = new WhiteSameNameRefDbm();
    private WhiteSameNameRefDbm() {}
    public static WhiteSameNameRefDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((WhiteSameNameRef)et).getSameNameRefId(), (et, vl) -> ((WhiteSameNameRef)et).setSameNameRefId(cti(vl)), "sameNameRefId");
        setupEpg(_epgMap, et -> ((WhiteSameNameRef)et).getSameNameId(), (et, vl) -> ((WhiteSameNameRef)et).setSameNameId(ctl(vl)), "sameNameId");
        setupEpg(_epgMap, et -> ((WhiteSameNameRef)et).getNextSameNameId(), (et, vl) -> ((WhiteSameNameRef)et).setNextSameNameId(ctl(vl)), "nextSameNameId");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    protected final Map<String, PropertyGateway> _efpgMap = newHashMap();
    { xsetupEfpg(); }
    @SuppressWarnings("unchecked")
    protected void xsetupEfpg() {
        setupEfpg(_efpgMap, et -> ((WhiteSameNameRef)et).getWhiteSameName(), (et, vl) -> ((WhiteSameNameRef)et).setWhiteSameName((OptionalEntity<WhiteSameName>)vl), "whiteSameName");
    }
    public PropertyGateway findForeignPropertyGateway(String prop)
    { return doFindEfpg(_efpgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "white_same_name_ref";
    protected final String _tableDispName = "white_same_name_ref";
    protected final String _tablePropertyName = "whiteSameNameRef";
    protected final TableSqlName _tableSqlName = new TableSqlName("white_same_name_ref", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnSameNameRefId = cci("same_name_ref_id", "same_name_ref_id", null, null, Integer.class, "sameNameRefId", null, true, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnSameNameId = cci("same_name_id", "same_name_id", null, null, Long.class, "sameNameId", null, false, false, true, "int8", 19, 0, null, false, null, null, "whiteSameName", null, null, false);
//"int64"
    protected final ColumnInfo _columnNextSameNameId = cci("next_same_name_id", "next_same_name_id", null, null, Long.class, "nextSameNameId", null, false, false, true, "int8", 19, 0, null, false, null, null, null, null, null, false);
//"int64"

    /**
     * same_name_ref_id: {PK, NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnSameNameRefId() { return _columnSameNameRefId; }
    /**
     * same_name_id: {NotNull, int8(19), FK to white_same_name}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnSameNameId() { return _columnSameNameId; }
    /**
     * next_same_name_id: {NotNull, int8(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNextSameNameId() { return _columnNextSameNameId; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnSameNameRefId());
        ls.add(columnSameNameId());
        ls.add(columnNextSameNameId());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnSameNameRefId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    /**
     * white_same_name by my same_name_id, named 'whiteSameName'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignWhiteSameName() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnSameNameId(), WhiteSameNameDbm.getInstance().columnSameNameId());
        return cfi("fk_white_same_name_ref", "whiteSameName", this, WhiteSameNameDbm.getInstance(), mp, 0, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "whiteSameNameRefList", false);
    }

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.WhiteSameNameRef"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.WhiteSameNameRefCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.WhiteSameNameRefBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<WhiteSameNameRef> getEntityType() { return WhiteSameNameRef.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public WhiteSameNameRef newEntity() { return new WhiteSameNameRef(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((WhiteSameNameRef)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((WhiteSameNameRef)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}

File: src/dbflute/adf/meta/whiteSameNameRefdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type WhiteSameNameRefDbm_T struct {
	df.BaseDBMeta
	ColumnSameNameRefId *df.ColumnInfo
	ColumnSameNameId *df.ColumnInfo
	ColumnNextSameNameId *df.ColumnInfo
}

func (b *WhiteSameNameRefDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *WhiteSameNameRefDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var WhiteSameNameRefDbm *WhiteSameNameRefDbm_T

func Create_WhiteSameNameRefDbm() {
	WhiteSameNameRefDbm = new(WhiteSameNameRefDbm_T)
	WhiteSameNameRefDbm.TableDbName = "white_same_name_ref"
	WhiteSameNameRefDbm.TableDispName = "white_same_name_ref"
	WhiteSameNameRefDbm.TablePropertyName = "whiteSameNameRef"
	WhiteSameNameRefDbm.TableSqlName = new(df.TableSqlName)
	WhiteSameNameRefDbm.TableSqlName.TableSqlName = "white_same_name_ref"
	WhiteSameNameRefDbm.TableSqlName.CorrespondingDbName = WhiteSameNameRefDbm.TableDbName

	var whiteSameNameRef df.DBMeta
	whiteSameNameRef = WhiteSameNameRefDbm
	WhiteSameNameRefDbm.DBMeta=&whiteSameNameRef
	sameNameRefIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo same_name_ref_id
	sameNameRefIdSqlName.ColumnSqlName = "same_name_ref_id"
	sameNameRefIdSqlName.IrregularChar = false
	WhiteSameNameRefDbm.ColumnSameNameRefId = df.CCI(&whiteSameNameRef, "same_name_ref_id", sameNameRefIdSqlName, "", "", "Integer.class", "sameNameRefId", "", true, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")
	sameNameIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo same_name_id
	sameNameIdSqlName.ColumnSqlName = "same_name_id"
	sameNameIdSqlName.IrregularChar = false
	WhiteSameNameRefDbm.ColumnSameNameId = df.CCI(&whiteSameNameRef, "same_name_id", sameNameIdSqlName, "", "", "Long.class", "sameNameId", "", false, false,true, "int8", 19, 0, "",false,"","", "whiteSameName","","",false,"int64")
	nextSameNameIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo next_same_name_id
	nextSameNameIdSqlName.ColumnSqlName = "next_same_name_id"
	nextSameNameIdSqlName.IrregularChar = false
	WhiteSameNameRefDbm.ColumnNextSameNameId = df.CCI(&whiteSameNameRef, "next_same_name_id", nextSameNameIdSqlName, "", "", "Long.class", "nextSameNameId", "", false, false,true, "int8", 19, 0, "",false,"","", "","","",false,"int64")

	WhiteSameNameRefDbm.ColumnInfoList = new(df.List)
	WhiteSameNameRefDbm.ColumnInfoList.Add(WhiteSameNameRefDbm.ColumnSameNameRefId)
	WhiteSameNameRefDbm.ColumnInfoList.Add(WhiteSameNameRefDbm.ColumnSameNameId)
	WhiteSameNameRefDbm.ColumnInfoList.Add(WhiteSameNameRefDbm.ColumnNextSameNameId)


	WhiteSameNameRefDbm.ColumnInfoMap=make(map[string]int)
	WhiteSameNameRefDbm.ColumnInfoMap["sameNameRefId"]=0
		WhiteSameNameRefDbm.ColumnInfoMap["sameNameId"]=1
		WhiteSameNameRefDbm.ColumnInfoMap["nextSameNameId"]=2
	    WhiteSameNameRefDbm.PrimaryKey = true
    WhiteSameNameRefDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &whiteSameNameRef
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(WhiteSameNameRefDbm.ColumnSameNameRefId)

	WhiteSameNameRefDbm.PrimaryInfo = new(df.PrimaryInfo)
	WhiteSameNameRefDbm.PrimaryInfo.UniqueInfo = ui
	
	var whiteSameNameRefMeta df.DBMeta = WhiteSameNameRefDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["WhiteSameNameRef"] = &whiteSameNameRefMeta
}
